File: Asmt.Api/Asmt.BL/DTOs/UserDto.cs
using System;

namespace Asmt.BL.DTOs;

/// <summary>
/// User DTO.
/// </summary>
public class UserDto
{   
    /// <summary>
    /// The user ID.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// The user name.
    /// </summary>
    public required string Name { get; set; }

    /// <summary>
    /// The user email.
    /// </summary>
    public required string Email { get; set; }

    /// <summary>
    /// The user role.
    /// </summary>
    public required string Role { get; set; }

    /// <summary>
    /// The user token.
    /// </summary>
    public string? Token { get; set; }
}

File: Asmt.App/src/app/dtos/user.dto.ts
/**
 * User DTO.
 */
export interface UserDto {
    /**
     * The user ID.
     */
    id: number;

    /**
     * The user name.
     */
    name: string;

    /**
     * The user email.
     */
    email: string;

    /**
     * The user role.
     */
    role: string;

    /**
     * The user token.
     */
    token?: string;
}
